formstack-baton-requests/src/main/scala/com/gu/identity/formstackbatonrequests/services/DynamoUpdateService.scala

```diff
@@ -27,32 +29,97 @@ case class DynamoUpdateService(
     }
   }
 
-  private def skipDecryptionError(err: Throwable): Either[Throwable, Unit] = {
-    err match {
-      case _: FormstackDecryptionError => Right(())
+  private def skipDecryptionError(formSubmissionError: Throwable): Either[Throwable, FormSubmissions] = {
+    formSubmissionError match {
+      case _: FormstackDecryptionError => Right(FormSubmissions(
+        submissions = List.empty,
+        pages = 0
+      ))
       case err => Left(err)
     }
   }
 
-  private def writeSubmissions(form: Form, lastUpdate: SubmissionTableUpdateDate, submissionPage: Int = 1, token: FormstackAccountToken): Either[Throwable, Unit] = {
-    formstackClient.formSubmissionsForGivenPage(submissionPage, form.id, lastUpdate, config.encryptionPassword, token) match {
-      case Left(err) => skipDecryptionError(err)
-      case Right(response) =>
-        logger.info(s"Received page $submissionPage of submissions out of ${response.pages} pages for form ${form.id}.")
-        val submissionsIdsWithEmails = submissionsWithEmailAndAccount(response.submissions, token.account)
-        logger.info(s"Writing ${submissionsIdsWithEmails.length} submission id(s) and emails to Dynamo")
-        dynamoClient.writeSubmissions(submissionsIdsWithEmails, config.bcryptSalt, config.submissionTableName) match {
-          case Right(batchWriteItemsResults) if batchWriteItemsResults.exists { result =>
-            val unprocessedItems = result.getUnprocessedItems
-            !unprocessedItems.isEmpty
-          } => Left(new Exception(s"Some items could not be written to DynamoDB: $batchWriteItemsResults"))
-          case Right(_) if submissionPage < response.pages => writeSubmissions(form, lastUpdate, submissionPage + 1, token)
-          case Right(_) => Right(())
-          case Left(err) => Left(err)
-        }
+  private def failUnprocessedItems(batchWriteItemsResults: Seq[BatchWriteItemResult]): Either[Throwable, Seq[BatchWriteItemResult]] ={
+    val unprocessedItems = batchWriteItemsResults.exists { result =>
+      val unprocessedItems = result.getUnprocessedItems
+      !unprocessedItems.isEmpty
```

Comment:
`unprocessedItems.nonEmpty` ?

Reply:
This is a `Map` so `nonEmpty` is not available.